### PR TITLE
add tested workaround for retrying DNS lookups

### PR DESCRIPTION
**Pull request by szuecs**
use workaround to handle https://github.com/golang/go/issues/23427

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -107,6 +107,21 @@ func main() {
...
+	stopChan := make(chan struct{})
+	// We need this to reliably fade on DNS change, which is right
+	// now not fixed with IdleConnTimeout in the http.Transport.
+	// https://github.com/golang/go/issues/23427
+	go func() {
+		for {
+			select {
+			case <-time.After(30 * time.Second):
+				kubeConfig.Transport.CloseIdleConnections()
+			case <-stopChan:
```
Comment (mikkeloscar) [inline main.go]: This channel is being read from twice, here and in the controller. We need two different channels or something like this, for this to work.

Comment (szuecs) [reply main.go]: see https://play.golang.org/p/M1QJsNKw4Pu we use it in skipper to shutdown >1 goroutines, too

**Comment by szuecs** [comment]
@mikkeloscar based on your comment here code showing that the shutdown handling works https://play.golang.org/p/M1QJsNKw4Pu . We use it in skipper to shutdown all goroutines from one channel.

File: main.go (hunk 1 above)
Comment (mikkeloscar) [reply main.go]: Ah you are right, we close the channel, I was thinking that you can't read twice!

**Comment by mikkeloscar** [comment]
:+1:

**Comment by aermakov-zalando** [comment]
Unfortunately, client-go complains if you try to use a custom transport:

    time="2018-06-15T14:46:06Z" level=fatal msg="using a custom transport with TLS certificate options or the insecure flag is not allowed"

**Pull request merged**